Feature/kickstarter summer 2019 (#557)
ordina-jworks/ordina-jworks.github.io: _posts/2019-10-25-Kickstarter-Trajectory-2019-Summer-Edition.md

During the 3 months kickstarter trajectory, there were 2 main periods:
* The first six weeks were filled with various trainings: security, backend, frontend, soft skill trainings,...
* The last, 2 DevCases were organized: Chatbot Dina and Zero Plastic Rivers

This blogpost will talk about both periods separately below. The first period is further separated into technical and non-technical trainings.

1.[Technical trainings](#technical-trainings)  
2.[Non-technical trainings](#non-technical-trainings)  
3.[DevCases](#devcases) 

## First Day
On our first arrival at Ordina, we were warmly welcomed by Anja and the Jworks team. 
We were introduced to the structure of Ordina and got to know some of the other Kickstarters. 
After the introduction, we were given a tour of the entire building, during which we met some of the colleagues at JWorks.
Four of the Kickstarters already did an internship at Ordina before coming to work officially, so they already knew a lot of the colleagues in Mechelen.
This didn't make any difference however, since everyone was very open and welcoming.

> Finally, we were given the keys to our cars, and our laptops, so that the next day we could get started with the courses, focused on both technical and soft skills.

Reviewer: Maybe rewrite the sentence as the overuse of commas (4) breaks the flow a lot. It also reads as if you got the keys to your laptops...
So maybe something like: 

> Finally, we got our laptops and the keys to our cars so that the next day we could get started with the courses which were focused on both technical and soft skills.

### Day 2
The second day, we built a small application to test the range of possibilities Spring Boot offers. 
The app was built to provide a system to save houses with their address, the inhabitants and some of the features it has like the type of garage.
It was a fun little project which taught us a lot of what Spring Boot can be used for. 
We've all used Spring Boot in our DevCase, so it was definitely worth following.

#### [Duncan Casteleyn](/author/duncan-casteleyn)
>What I really liked about Ken's Spring Boot session, was that we did a lot of hands-on live coding.
This made the course very interactive, providing possibilities for both feedback to Ken as input of our own.
I learned a lot from this session, even though I already had a lot of prior knowledge of Spring Boot.
The theoretical start on Spring was a very useful refresh to get up to speed again.

## Docker
The Docker course, given by [Tom Verelst](/author/tom-verelst), started with a theoretical explanation about the advantages of containerization, the difference with virtual machines and how a container is built. 
The main part of the course was a hands-on session. 

> We learned to create an image of our application and push it to Docker hub, spin up a docker container and write efficient Docker compose files.

Reviewer: For consistency: docker -> Docker ?

#### [Brecht Porrez](/author/brecht-porrez)
>During this training I really experienced the advantages of Docker. 
I now use it almost daily during the Zero Plastic Rivers DevCase.

> For example, if I need a test database, I quickly start up a docker container.

Reviewer: For consistency: docker -> Docker ?

#### [Nicholas Meyers](/author/nicholas-meyers)
>I’m very interested in how applications are built and deployed, which is why I found the Kubernetes session very interesting. 
I’d love to learn more about this technology in the future. 
The hands-on way of working helped me a lot, because this was quite new to me, which made it not the easiest course to follow.

## Test-Driven Development

In the DevOps track, we received an interesting lesson about Test Driven Development from [Pieter Van Hees](/author/pieter-van-hees).
In school, TDD is usually not taught and testing comes after developing. 
What TDD aims to do is speeding up the development process by thinking about what exactly you want your program to do and which exact results you want.
After pouring those requirements into unit tests, you can start developing and immediately testing whatever you wrote. 
There are many advantages of working with TDD, but it mainly makes it easier later on in the development process. 
In the beginning, there's more work involved because you need to write all the tests. 
In the long run however, it saves a lot of time because you can immediately spot mistakes using your unit tests.

> Another good use case is refactoring code. With the test already in place, you have a great test to see if you did the refactoring correct.

Reviewer: 1 sentence per line

This is why Michaëla introduced us to Agile which helped us to communicate and work better as a team. 

She did this by means of a productivity game.
The game worked as follows: 
* Everyone stands in a circle and the group gets one (small) ball. 
* The team was to throw around the ball during 2 minutes, while a metronome was ticking in the background. 
* Every time the ball got caught on a tick, one task was completed.
* After 2 minutes, the team got 30 seconds to decide on a new strategy, but were only allowed to change 1 thing at a time (an extra amount of balls, a different way of throwing, reverting back to a previous way of working,...)

Playing this game for 6 rounds, the productivity went up exponentially. The team had matters in its own hands, which made them think for themselves.
At the end we refreshed a couple of famous agile practices such as the SCRUM framework, which is a popular way of working together to quickly and reliably release new features.

#### [Lore Vanderlinden](/author/lore-vanderlinden)
>The agile session was a very enriching experience. We learned the basic concepts of agile the right way. 

> Michaele was a very inspiring agile coach, making the learning process easier by using a hands-on way of teaching. 

Reviewer: Michaele -> Michaëla

#### [Jasper Rosiers](/author/jasper-rosiers)
>What I found most interesting about the Scrum framework and the Agile way of working, is that there’s many moments to reflect on how the work is going and how well the team is working together. 
The daily scrum is a very powerful moment, which made us use it in our Chatbot DevCase. 
Frequent meetings with the Product Owner and keeping him close to the project is another aspect I love about the Scrum framework. 
I will definitely look more into it in the future, since I'm aiming to become a Scrum Master.

<div style="text-align: center;">
  <img src="/img/2019-10-25-Kickstarter-Trajectory-2019-Summer-Edition/ScrumLayout.jpg" width="80%" height="80%" target="_blank">
</div>

## Agile Hands-On

[Michaëla Broeckx](/author/michaela-broeckx) gave us an introduction to how Agile development works in practice. 
It was a very hands-on session that helped us gain more knowledge and experience in the world of Agile development. 
First, we saw how the waterfall method worked, but then quickly noticed it wasn’t perfect and had a lot of flaws. 
This is why Michaëla introduced us to Agile which helped us to communicate and work better as a team. 

> She did this by means of a productivity game.

Reviewer 2: I think Michaëla made some photos. Would be cool to include one of them. :)

The game worked as follows: 
* Everyone stands in a circle and the group gets one (small) ball. 

> * The team was to throw around the ball during 2 minutes, while a metronome was ticking in the background. 

Reviewer 2: ```suggestion
* The team was to throw around the ball during two minutes, while a metronome was ticking in the background. 
```

* Every time the ball got caught on a tick, one task was completed.

> * After 2 minutes, the team got 30 seconds to decide on a new strategy, but were only allowed to change 1 thing at a time (an extra amount of balls, a different way of throwing, reverting back to a previous way of working,...)

Reviewer 2: ```suggestion
* After two minutes, the team got 30 seconds to decide on a new strategy, but were only allowed to change one thing at a time (an extra amount of balls, a different way of throwing, reverting back to a previous way of working,...)
```

Playing this game for 6 rounds, the productivity went up exponentially. The team had matters in its own hands, which made them think for themselves.
At the end we refreshed a couple of famous agile practices such as the SCRUM framework, which is a popular way of working together to quickly and reliably release new features.

#### [Lore Vanderlinden](/author/lore-vanderlinden)
>The agile session was a very enriching experience. We learned the basic concepts of agile the right way. 
Michaëla was a very inspiring agile coach, making the learning process easier by using a hands-on way of teaching. 
She used real life examples to show us the advantages of working in an agile manner.

# Final day

On the day after the final course day, all the Kickstarters gave a short presentation about themselves in front of the others and the management. 
This way, everyone present got to know the others, with both their professional interests and achievements, as well as a little on the personal side. 
Afterwards, there was a moment for networking and socializing with everyone, and an official graduation. 

> The next day, the DevCases started, which we’ll explain below!

Reviewer 2: ```suggestion
The next day, the dev cases started, which we’ll explain below!
```

<div style="text-align: center;">
  <img src="/img/2019-10-25-Kickstarter-Trajectory-2019-Summer-Edition/FinalDay.png" width="80%" height="80%" target="_blank">
</div>

# DevCases 

> The team of 8 JWorks Kickstarters was divided into 2 teams, and thus 2 DevCases:

Reviewer 2: ```suggestion
The team of 8 JWorks Kickstarters was divided into two teams, and thus 2 dev cases:
```

The main objective of the application is to create a monitoring network to collect plastic waste, for example, in dams, locks or water treatment plants. 
This way, plastic flows can be calculated for example in sub-basins or piers to estimate the total flow to the estuary.
To activate this system, plastic bottles in the Schelde river will be released at different strategic points with GPS trackers and personalized labels. 
These contain relevant information such as the identifier or the url to the application.

The application consists of two parts.
The first part is aimed at citizens who wish to help the cause, who can notify this surveillance network when they find a bottle as shown in the image on the right.
The second part is aimed at the researchers, and could be seen as the "backend" of the project, where the data given by the GPS trackers and the citizens is visualized in a clear and orderly way.

## Chatbot Dina

In the second DevCase, we built chatbot Dina for the Fleet department of Ordina. 
The Chatbot team set off using the Chatlayer bot framework, later to be joined by an implementation in Dialogflow. 

> Since chatbots are a relatively new technology, we wanted to keep options open and look for the best possible implementation. 

Reviewer 2: ```suggestion
Since chatbots are a relatively new technology, we wanted to keep our options open and look for the best possible implementation. 
```
